server_test.go

```diff
@@ -141,6 +145,81 @@ var _ = Describe("Server", func() {
+			})
+		})
+
+		Context("when the client requests the list of prompts with an invalid cursor", func() {
+			It("responds with a protocol error", func() {
+				stdin.WriteString(`{"jsonrpc":"2.0","id":2,"method":"prompts/list","params":{"cursor":"invalid-cursor"}}`)
+				Eventually(session.Out).Should(gbytes.Say("\n"))
+				Expect(lastResponse(session.Out.Contents())).To(MatchJSON(`{"jsonrpc":"2.0","error":{"code":-32602,"message":"Invalid params"},"id":2}`))
+			})
+		})
+
+		Context("when the client gets a prompt", func() {
+			It("processes the prompt with the provided arguments", func() {
+				stdin.WriteString(`{"jsonrpc":"2.0","id":2,"method":"prompts/get","params":{"name":"example","arguments":{"text":"test input"}}}`)
+				Eventually(session.Out).Should(gbytes.Say("\n"))
+
+				responses := allResponses(session.Out.Contents())
+				Expect(responses).To(HaveLen(4)) // First two are initialization, last two are the prompt response
+				Expect(responses[len(responses)-2]).To(MatchJSON(`{"jsonrpc":"2.0","method":"test/notification","params":{"message":"Processing text"}}`))
+				Expect(responses[len(responses)-1]).To(MatchJSON(`{"jsonrpc":"2.0","id":2,"result":{"messages":[{"role":"assistant","content":{"type":"text","text":"Processed: test input"}}]}}`))
+			})
+
+			// Missing arguments error
+			Context("when required arguments are missing", func() {
+				It("responds with an error", func() {
+					stdin.WriteString(`{"jsonrpc":"2.0","id":2,"method":"prompts/get","params":{"name":"example","arguments":{}}}`)
+					Eventually(session.Out).Should(gbytes.Say("\n"))
+					Expect(lastResponse(session.Out.Contents())).To(MatchJSON(`{"jsonrpc":"2.0","error":{"code":-32602,"message":"Missing required argument: text"},"id":2}`))
+				})
+			})
+
+			// Unknown prompt error
+			Context("when the prompt doesn't exist", func() {
+				It("responds with an error", func() {
+					stdin.WriteString(`{"jsonrpc":"2.0","id":2,"method":"prompts/get","params":{"name":"nonexistent","arguments":{}}}`)
+					Eventually(session.Out).Should(gbytes.Say("\n"))
+					Expect(lastResponse(session.Out.Contents())).To(MatchJSON(`{"jsonrpc":"2.0","error":{"code":-32602,"message":"Unknown prompt: nonexistent"},"id":2}`))
+				})
+			})
+
+			// Prompt processing error
+			Context("when the prompt processing fails", func() {
+				It("responds with a prompt error", func() {
+					stdin.WriteString(`{"jsonrpc":"2.0","id":2,"method":"prompts/get","params":{"name":"example","arguments":{"text":""}}}`)
+					Eventually(session.Out).Should(gbytes.Say("\n"))
+					Expect(lastResponse(session.Out.Contents())).To(MatchJSON(`{"jsonrpc":"2.0","id":2,"result":{"messages":[{"role":"assistant","content":{"type":"text","text":"input text cannot be empty"}}]}}`))
```

Comment:
It doesn't seem right to me to include the validation or rate limit errors in the generated prompt that is returned. Seems like these should be actual errors?

Do you think the existing tool rate limit error should be surfaced as a JSON-RPC error rather than a tool execution error?